examples/scripts/04-ppo-sentiment-accelerate.py

```diff
@@ -0,0 +1,179 @@
+# its own dataset.
+def build_dataset(config):
+    """
+    Build dataset for training. This builds the dataset from `load_dataset`, one should 
+    customize this function to train the model on its own dataset.
+    
+    Args:
+        dataset_name (`str`): 
+            The name of the dataset to be loaded.
+    
+    Returns:
+        dataloader (`torch.utils.data.DataLoader`):
+            The dataloader for the dataset.
+    """
+    dataset_name = config["dataset_name"]
+    tokenizer = AutoTokenizer.from_pretrained(config["model_name"])
+    tokenizer.pad_token = tokenizer.eos_token
+    # load imdb with datasets
+    ds = load_dataset(dataset_name, split='train')
+    ds = ds.rename_columns({'text': 'review', 'label': 'sentiment'})
+    ds = ds.filter(lambda x: len(x["review"])>200, batched=False)
+
+    input_size = LengthSampler(config["txt_in_min_len"], config["txt_in_max_len"])
+
+    def tokenize(sample):
+        sample["tokens"] = tokenizer.encode(sample["review"])[:input_size()]
+        sample["query"] = tokenizer.decode(sample["tokens"])
+        return sample
+
+    ds = ds.map(tokenize, batched=False)
+
+    def collater(data):
+        return dict((key, [d[key] for d in data]) for key in data[0])
+
+    dataloader = torch.utils.data.DataLoader(ds, batch_size=config['batch_size'], collate_fn=collater)
+    return dataloader
+
+# We retrieve the dataloader by calling the `build_dataset` function.
+dataloader = build_dataset(config)
+
+# Now let's build the model, the reference model, and the tokenizer.
+model = AutoModelForCausalLMWithValueHead.from_pretrained(config["model_name"])
+ref_model = AutoModelForCausalLMWithValueHead.from_pretrained(config["model_name"])
+tokenizer = AutoTokenizer.from_pretrained(config["model_name"])
+
+tokenizer.pad_token = tokenizer.eos_token
+
+# We then build the PPOTrainer, passing the model, the reference model, the tokenizer
+ppo_trainer = PPOTrainer(model, ref_model, tokenizer, dataloader, **config)
+
+# the PPOTrainer has a dataloader attribute, which we can use to get the dataloader - 
+# this step is important in a distributed setting, as the dataloader needs to be
+# converted to a distributed dataloader.
+dataloader = ppo_trainer.dataloader
+
+# We then build the sentiment analysis pipeline, passing the model name and the
+# sentiment analysis pipeline arguments. Let's also make sure to set the device
+# to the same device as the PPOTrainer.
+device = ppo_trainer.accelerator.device
+if device.index is None:
+    # single GPU - maybe introduce this hack inside PPOTrainer?
+    device = 0
+sentiment_pipe = pipeline("sentiment-analysis","lvwerra/distilbert-imdb", device=device)
+
+# We then define the arguments to pass to the `generate` function. These arguments
+# are passed to the `generate` function of the PPOTrainer, which is a wrapper around 
+# the `generate` function of the trained model.
+gen_kwargs = {
```

Comment:
Might be cleaner to use the new `GenerationConfig` in `transformers` here: https://huggingface.co/docs/transformers/v4.25.1/en/main_classes/text_generation#transformers.GenerationConfig

Reply:
Agreed!
However I tried to run `generate` with `GenerationConfig` and it seems that the feature is currently only available on the `main` branch. This PR: https://github.com/huggingface/transformers/pull/20388/ has been merged 2 weeks ago
So maybe let's address this in a follow up PR !

Reply:
Yes, let's not commit to very new features, otherwise we need very hard `transformers` dependency